base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
+end
```

Comment:
- [ ] Kindly add `attr_accessor` for the `@nameable` variable👍

Reply:
@ha-manel 
Please explain why I have to add `attr_accessor` because I can't think of a scenario where I would have to or even want to access `@nameable`